### PR TITLE
Feature: Autoload sample to synth or drum when LOAD is toggled On while previewing sounds in sound browser

### PR DESCRIPTION
**Pull request by soymonitus**
I find loading samples to instruments pretty slow specially when I want to try a lot of different sounds. This feature allows you to tap Load to engage "autoload" while browsing samples for a synth or drum (not available for audioclips).
So while Load is lighted, the preview action when scrolling through samples will also load the sample as if you had pressed Enter and then opened the browser.

TO-DO:

- [x] Fix the issue with golden knobs jumping from Filter cutoff to Custom2

https://github.com/user-attachments/assets/b1f56442-fa41-4248-94e7-453942e4ede7

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,9 @@
 - Added Vibrato and Sidechain patch cables to Automation View Overview and Grid Shortcuts
 - Added ability to enter clips in `Song Grid View Green Mode` by `Pressing a Clip Pad` + `Pressing the Clip button` if you have `Select in Green Mode` enabled in the `SETTINGS > DEFAULTS > UI > SONG > GRID` menu.
 
+#### <ins>Instrument View</ins>
```
Comment (seangoodvibes) [inline CHANGELOG.md:23]: Just a nit, call it Instrument Clips

Comment (soymonitus) [reply CHANGELOG.md:23]: Ok, by the way. What do you think about changing from Learn to Load as the toggle button?
In my faceplate i have Learn labels as Learn/Copy so i related it to "copying the sample to the synth", but maybe Load is much more obvious for a user

Comment (soymonitus) [reply CHANGELOG.md:23]: tried with LOAD for the toggle and feels nice and intuitive

Comment (soymonitus) [reply CHANGELOG.md]: Ok changes addressed and bonus point: fixed issue with SCW not loading as that when osc1type is already wavetable

**Comment by soymonitus** [comment]
@seangoodvibes this is ready for review!

File: docs/community_features.md [hunk 2]
```diff
@@ -908,18 +908,22 @@ to each individual note onset. ([#1978])
...
 - ([#2641]) Added new note and note row editor menu's to edit note and note row parameters.
   - Hold a note and press the select encoder to enter the note editor menu. While in the note editor menu, the selected note will blink. You can select other notes by pressing the notes on the grid.
   - Hold a note row audition pad and press the select encoder to enter the note row editor menu. While in the note row editor menu, the select note row audition pad will blink. You can select other note row's by pressing the note row audition pad.
 
+#### 4.3.12 - Auto Load sample when browsing
+
+- ([#2676]) When browsing for sample to assign to a Synth or to a Kit Row, you can now enable auto-load of previewed samples. To do this, open the sample browser (`SHIFT + BROWSE`), and engage `AUTO-LOAD` by pressing `LOAD`. While `LOAD` LED is on, all the samples that you preview will automatically be loaded to the instrument as if you had confirmed the selection. This makes previewing single cycle waveforms way more convenient, for example.
```
Comment (seangoodvibes) [inline docs/community_features.md:925]:
```suggestion
- ([#2676]) When browsing for sample to assign to a Synth or to a Kit Row, you can now enable auto-load of previewed samples. To do this, open the sample browser (`SHIFT + BROWSE`), and engage `AUTO-LOAD` by pressing `LOAD`. While the `LOAD` LED is on, all the samples that you preview will automatically be loaded to the instrument as if you had confirmed the selection. This makes previewing single cycle waveforms way more convenient, for example.
```

File: src/deluge/gui/ui/browser/sample_browser.cpp [hunk 3]
```diff
@@ -468,6 +471,14 @@ ActionResult SampleBrowser::buttonAction(deluge::hid::Button b, bool on, bool in
 		}
 	}
 
+	// Learn button: toggle autoload (only for non-audioclips)
```
Comment (seangoodvibes) [inline src/deluge/gui/ui/browser/sample_browser.cpp:474]:
```suggestion
	// Load button: toggle auto-load (only for non-audio clips)
```

File: src/deluge/gui/ui/browser/sample_browser.cpp [hunk 4]
```diff
@@ -468,6 +471,14 @@ ActionResult SampleBrowser::buttonAction(deluge::hid::Button b, bool on, bool in
...
+	else if (b == LOAD && getCurrentClip()->type != ClipType::AUDIO) {
+		if (!on) {
```
Comment (seangoodvibes) [inline src/deluge/gui/ui/browser/sample_browser.cpp:476]: was "(!on)" intentional here? typically I've seen (on)

File: src/deluge/gui/ui/browser/sample_browser.cpp [hunk 5]
```diff
@@ -562,6 +573,12 @@ void SampleBrowser::previewIfPossible(int32_t movementDirection) {
 
 		AudioEngine::previewSample(&filePath, &currentFileItem->filePointer, shouldActuallySound);
 
+		if (autoLoadEnabled && getCurrentClip()->type != ClipType::AUDIO) {
+			// Feature: if Load has been toggled on, then the file will be auto-loaded into the current instrument
+			// as if you had confirmed with Select encoder, but keeping the browser open.
```
Comment (seangoodvibes) [inline src/deluge/gui/ui/browser/sample_browser.cpp:578]:
```suggestion
			// as if you had confirmed with the Select encoder, but keeping the browser open.
```

File: src/deluge/gui/ui/browser/sample_browser.cpp (hunk 4 above)
Comment (soymonitus) [reply src/deluge/gui/ui/browser/sample_browser.cpp:476]: Yes, so it works on release, for consistency, just like with toggling scale mode in clips for example

**Review by seangoodvibes — APPROVED** [review]

**Pull request merged**